server/routes/allocate-to-activity/handlers/allocationDashboard.ts

```diff
@@ -129,90 +128,50 @@ export default class AllocationDashboardRoutes {
-      .then(inmates =>
-        inmates.filter(
-          i =>
-            filters.incentiveLevelFilter === 'All Incentive Levels' ||
-            !suitableIeps ||
-            suitableIeps.includes(i.currentIncentive?.level.description),
-        ),
-      )
-      .then(inmates => this.filterByWra(inmates, filters, suitableWpas))
-      .then(inmates =>
-        inmates.filter(
-          i =>
-            !candidateQueryLower ||
-            i.prisonerNumber.toLowerCase().includes(candidateQueryLower) ||
-            `${i.firstName} ${i.lastName}`.toLowerCase().includes(candidateQueryLower),
-        ),
-      )
-
-    const prisonerNumbers = pageOfInmates.map(i => i.prisonerNumber)
-    const [currentAllocations, educationLevels]: [PrisonerAllocations[], Education[]] = await Promise.all([
-      this.activitiesService.getPrisonerAllocations(user.activeCaseLoad.caseLoadId, prisonerNumbers, user),
-      this.prisonService.getEducations(prisonerNumbers, user),
-    ])
+    const suitableIeps = this.getSuitableIepLevels(filters)
+    const suitableWpas = this.getSuitableRiskLevelCodes(filters)
+    const suitableForEmployed = this.getSuitableForEmployed(filters)
 
-    return pageOfInmates
-      .map(inmate => {
-        const otherAllocations =
-          currentAllocations
-            .find(a => a.prisonerNumber === inmate.prisonerNumber)
-            ?.allocations.map(a => ({
-              id: a.scheduleId,
-              scheduleName: a.scheduleDescription,
-              isUnemployment: a.isUnemployment,
-            })) || []
-
-        const inWork = otherAllocations.filter(a => !a.isUnemployment).length > 0
-
-        return {
-          name: `${inmate.firstName} ${inmate.lastName}`,
-          prisonerNumber: inmate.prisonerNumber,
-          cellLocation: inmate.cellLocation,
-          otherAllocations,
-          inWork,
-          releaseDate: inmate.releaseDate ? parseDate(inmate.releaseDate) : null,
-          educationLevels: educationLevels.filter(e => e.bookingId.toString() === inmate.bookingId),
-        }
-      })
-      .filter(
-        i =>
-          !filters.employmentFilter ||
-          filters.employmentFilter === 'Everyone' ||
-          (filters.employmentFilter === 'In work' && i.inWork) ||
-          (filters.employmentFilter === 'Not in work' && !i.inWork),
-      )
+    return this.activitiesService
+      .getActivityCandidates(scheduleId, user, suitableIeps, suitableWpas, suitableForEmployed, candidateQueryLower)
+      .then(candidates => candidates.map(c => ({ ...c, releaseDate: c.releaseDate ? parseDate(c.releaseDate) : null })))
   }
 
-  private filterByWra = (inmates: Prisoner[], filters: Filters, suitableWras: string[]): Prisoner[] => {
+  private getSuitableIepLevels = (filters: Filters): string[] => {
+    if (filters.incentiveLevelFilter === 'All Incentive Levels') {
+      return undefined
+    }
+
+    return filters.incentiveLevelFilter?.split(' or ')
```

Comment:
I think changing the option values to enum strings and using a delimiter character such as `|` could be safer here. So the value would be something like, `LOW|MEDIUM|HIGH`.